google/cloud/storage/google_cloud_cpp_storage.cmake

```diff
@@ -235,6 +237,7 @@ target_link_libraries(
            absl::time
            absl::variant
            google-cloud-cpp::common
+           google-cloud-cpp::rest-internal
```

Comment:
This means that any time `storage` is enabled (via `GOOGLE_CLOUD_CPP_ENABLE`) we also need to set `GOOGLE_CLOUD_CPP_ENABLE_REST`:

https://github.com/googleapis/google-cloud-cpp/blob/8a2298482d4ecbdcbc3406afa65c3d4f44b1d749/google/cloud/CMakeLists.txt#L47


That SGTM.  I wonder if we should detect an incompatible setting. For example, we could issue a fatal error or at least a warning if somebody explicitly sets `-DGOOGLE_CLOUD_CPP_ENABLE=storage -DGOOGLE_CLOUD_CPP_ENABLE_REST=OFF` in their CMake configuration.